src/execution_plan/optimizations/apply_join.c

```diff
@@ -166,14 +170,28 @@ void applyJoin(ExecutionPlan *plan) {
 			} else {
 				// The Cartesian Product still has a child operation; introduce the join op as another child.
 				ExecutionPlan_AddOp(cp, value_hash_join);
+				// Search for additional filters which can be solved by the join operation branches.
+				for(int k = j + 1; k < filter_count; k++) {
+					/* Fix for issue #869 https://github.com/RedisGraph/RedisGraph/issues/869.
+					 * When trying to replace a cartesian product which followed by multiple filters that can be resolved by the same two branches
+					 * the application crashed on assert(lhs_resolving_stream != rhs_resolving_stream) since before the fix only one filter was used
+					 * to create the hash join, and the other were ignored. */
+					// Check for additional filters which checking for the same equality.
+					OpFilter *additional_filter = filter_ops[k];
+					FT_FilterNode *additional_filter_tree = additional_filter->filterTree;
+
+					rax *references = FilterTree_CollectModified(additional_filter_tree);
+					OpBase *op = ExecutionPlan_LocateReferences(cp, NULL, references);
+					if(op) {
+						ExecutionPlan_RemoveOp(plan, (OpBase *)additional_filter);
+						ExecutionPlan_PushBelow(op, (OpBase *)additional_filter);
+					}
+					raxFree(references);
```

Comment:
No need to check for `if(op)`
You're guarantee that `filter` will be either placed as a parent of `cp` or as a parent of one of the `hashjoin` operations.

Reply:
removed